### PR TITLE
JAVA-252: Fixing failing tests on ML 10 nightly

### PR DESCRIPTION
**Pull request by rjrudin**
I looked for an annotation-based approach, but that's not easily done with JUnit 4. So just putting an `if` statement at the start of each RowManagerTest test that should only run on ML 11 or higher.

### CODE REVIEW

File: marklogic-client-api/src/test/java/com/marklogic/client/test/MarkLogicVersion.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package com.marklogic.client.test;
```
Comment (rjrudin) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/MarkLogicVersion.java:1]: This is a variation of the DHF class with the same name that DHF uses to conditionally run tests. Cleaned it up a bit and added a bunch of tests to verify that it's able to parse different versions correctly.

File: marklogic-client-api/src/test/java/com/marklogic/client/test/RowManagerTest.java [hunk 2]
```diff
@@ -1489,11 +1492,14 @@ public void testBucketGroup() {
...
   public void testFromDocUrisWithWordQuery() {
+    if (!markLogicIsVersion11OrHigher()) {
+      return;
+    }
+
     RowManager rowMgr = Common.client.newRowManager();
     PlanBuilder p = rowMgr.newPlanBuilder();
     PlanBuilder.ExportablePlan builtPlan = p.fromDocUris(p.cts.wordQuery("trumpet"), "");
 
-    RowSet<RowRecord> recordRowSet = rowMgr.resultRows(builtPlan);
```
Comment (rjrudin) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/RowManagerTest.java:1496]: Unused row, removed

File: marklogic-client-api/src/test/java/com/marklogic/client/test/RowManagerTest.java [hunk 3]
```diff
@@ -1520,7 +1529,6 @@ public void testFromDocUrisWithDirectoryQuery() throws Exception {
     batcher.addAs("/testFromDocUrisWithDirectoryQueryNew/doc4.txt", meta, "This is doc4");
     batcher.flushAndWait();
 
-    RowSet<RowRecord> recordRowSet = rowMgr.resultRows(builtPlan);
```
Comment (rjrudin) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/RowManagerTest.java:1523]: Unused row, removed

**Review by BillFarber — APPROVED** [review]
LGTM

**Review by llinggit — APPROVED** [review]

**Pull request merged**